src/bus/index.spec.js

```diff
@@ -0,0 +1,265 @@
+    this.sandbox = sandbox.create();
+
+    baseRequest = {
+      delete: this.sandbox.stub().resolves(),
+      get: this.sandbox.stub().resolves(),
+      post: this.sandbox.stub().resolves(),
+      put: this.sandbox.stub().resolves()
+    };
+
+    baseSdk = {
+      config: {
+        audiences: {
+          bus: fixture.build('audience')
+        }
+      }
+    };
+  });
+
+  afterEach(function() {
+    this.sandbox.restore();
+  });
+
+  describe('constructor', function() {
+    let bus;
+
+    beforeEach(function() {
+      bus = new Bus(baseSdk, baseRequest);
+    });
+
+    it('sets a base url for the class instance', function() {
+      expect(bus._baseUrl).to.equal(`${baseSdk.config.audiences.bus.host}`);
+    });
+
+    it('appends the supplied request module to the class instance', function() {
+      expect(bus._request).to.equal(baseRequest);
+    });
+
+    it('appends the supplied sdk to the class instance', function() {
+      expect(bus._sdk).to.equal(baseSdk);
+    });
+
+    it('appends an instance of Fields to the class instance', function() {
+      expect(bus.channels).to.be.an.instanceof(Channels);
+    });
+
+    it("returns a function called 'connect'", function() {
+      expect(bus.connect).to.be.a('function');
+    });
+  });
+
+  describe('connect', function() {
+    let expectedHost;
+    let expectedOrganization;
+    let server;
+
+    beforeEach(function() {
+      expectedHost = `wss://${faker.internet.domainName()}`;
+      expectedOrganization = fixture.build('organization');
+
+      server = new Server(
+        `${expectedHost}/organizations/${expectedOrganization.id}/stream`
+      );
+    });
+
+    afterEach(function() {
+      server.stop();
+    });
+
+    context('when a websocket already exists for the organization', function() {
+      let bus;
+      let expectedSocket;
+      let promise;
+      let sdk;
+      let socket;
+
+      beforeEach(function() {
+        sdk = {
+          ...baseSdk,
+          auth: {
+            ...baseSdk.auth,
+            getCurrentApiToken: this.sandbox.stub().resolves()
+          }
+        };
+
+        bus = new Bus(sdk, baseRequest);
+        bus._baseSocketUrl = expectedHost;
+
+        socket = new WebSocket(
+          `${expectedHost}/organizations/${expectedOrganization.id}/stream`
+        );
+        expectedSocket = new Socket(socket, expectedOrganization.id);
+
+        bus._sockets[expectedOrganization.id] = expectedSocket;
+
+        promise = bus.connect(expectedOrganization.id);
+      });
+
+      it('fulfills the promise', function() {
+        expect(promise).to.be.fulfilled;
+      });
+
+      it('does not fetch an api token', function() {
+        return promise.then(() => {
+          expect(sdk.auth.getCurrentApiToken).to.not.be.called;
+        });
+      });
+
+      it('resolves the promise with the existing socket', function() {
+        return promise.then((socket) => {
+          expect(socket).to.deep.equal(expectedSocket);
```

Comment:
1. Just `expect(socket).to.deep.equal(expectedSocket);` as you want to be sure that it is the exact same socket, not a similar one
2. This should be enough expect for this test